spec/lib/billing_facade_client_spec.rb

```diff
@@ -11,7 +11,52 @@
   before do
     BillingFacadeClient.site=billing_url
   end
-  
+
+  describe 'CostForModule' do
+    let(:product_name) { 'product'}
+    let(:module_name) { 'module name'}
+    let(:cost_code) { 'a cost code'}
+    let(:product_name) { 'a product name' }
+    context '#validate_response_cost_for_module_name' do
+      it 'returns false if the program do not match' do
+        response = { cost_code: cost_code, 'product': 'random', 'module': module_name}
+        expect(BillingFacadeClient.validate_response_cost_for_module_name(response, 
+          product_name, module_name, cost_code)).to eq(false)
+      end
+
+      it 'returns false if the cost code do not match' do
+        response = { cost_code: 'random', 'product': product_name, 'module': module_name}
+        expect(BillingFacadeClient.validate_response_cost_for_module_name(response, 
+          product_name, module_name, cost_code)).to eq(false)
+      end
+      it 'returns false if the module name do not match' do
+        response = { 'module': 'random', 'product': product_name, cost_code: cost_code}
+        expect(BillingFacadeClient.validate_response_cost_for_module_name(response, 
+          product_name, module_name, cost_code)).to eq(false)        
+      end
+      it 'returns false if the response has some errors' do
+        response = { errors: []}
+        expect(BillingFacadeClient.validate_response_cost_for_module_name(response, 
+          product_name, module_name, cost_code)).to eq(false)
+      end
+      it 'returns false if the response does not have a price' do
```

Comment:
I think the responses for the above tests should include prices, as the lack of price would cause them to return false regardless of their specific tests.